routes/api/users/index.js

```diff
@@ -53,6 +53,18 @@ router.get('/:username', async (req, res) => {
         res.status(400).json({error: 'Failed to retrieve user'});
     }
 });
+/**
+ * Check if username is taken
+ */
+router.get('/:username/exists', async (req, res) => {
+   const username = req.params.username; 
```

Comment:
You should ensure that `toLowerCase` doesn't error out if this is null with `const username = req.params.username || '';`

Reply:
Ah okay, I was thinking about adding that but I thought that it couldn't reach this stage if username is empty.

I guess adding it is a much safer option though to completely avoid the possibility of errors in case something does happen. I'll add that!